src/main/java/uk/gov/justice/laa/amend/claim/forms/validators/AllowedTotalsValueValidator.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.justice.laa.amend.claim.forms.validators;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import lombok.AllArgsConstructor;
+import uk.gov.justice.laa.amend.claim.forms.AllowedTotalForm;
+import uk.gov.justice.laa.amend.claim.forms.MonetaryValueForm;
+import uk.gov.justice.laa.amend.claim.forms.annotations.ValidAllowedTotal;
+import uk.gov.justice.laa.amend.claim.forms.annotations.ValidMonetaryValue;
+import uk.gov.justice.laa.amend.claim.models.ClaimField;
+
+import java.math.BigDecimal;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+@AllArgsConstructor
+public class AllowedTotalsValueValidator extends Validator implements ConstraintValidator<ValidAllowedTotal, AllowedTotalForm> {
```

Comment:
Could probably extract this into a separate `MoneyValidator` as a lot of the logic is shared, but happy for this to be done separately in the interest of getting this merged.